web/template/layout/pagination.gotmpl

```diff
@@ -1,30 +1,41 @@
-    <p class="govuk-visually-hidden" aria-labelledby="pagination-label">Pagination navigation</p>
-    <p class="moj-pagination__results">Showing <b>{{ .Start }}</b> to <b>{{ .End }}</b> of <b>{{ .TotalItems }}</b> cases</p>
+{{ define "pagination-inner" }}
+  <p class="moj-pagination__results">Showing <b>{{ .Start }}</b> to <b>{{ .End }}</b> of <b>{{ .TotalItems }}</b> cases</p>
 
-    {{ if ne 1 .TotalPages }}
-      <ul class="moj-pagination__list">
-        {{ if .HasPrevious }}
-          <li class="moj-pagination__item  moj-pagination__item--prev">
-            <a class="moj-pagination__link" href="?page={{ .PreviousPage }}">Previous<span class="govuk-visually-hidden"> page</span></a>
-          </li>
-        {{ end }}
+  {{ if ne 1 .TotalPages }}
+    <ul class="moj-pagination__list">
+      {{ if .HasPrevious }}
+        <li class="moj-pagination__item  moj-pagination__item--prev">
+          <a class="moj-pagination__link" href="{{ .Query }}page={{ .PreviousPage }}">Previous<span class="govuk-visually-hidden"> page</span></a>
+        </li>
+      {{ end }}
 
-        {{ range .Pages }}
-          {{ if eq . $.CurrentPage }}
-            <li class="moj-pagination__item moj-pagination__item--active"><span class="govuk-visually-hidden">Page </span>{{ . }}<span class="govuk-visually-hidden"> of {{ $.TotalPages }}</span></li>
-          {{ else }}
-            <li class="moj-pagination__item"><a class="moj-pagination__link" href="?page={{ . }}"><span class="govuk-visually-hidden">Page </span>{{ . }}<span class="govuk-visually-hidden"> of {{ $.TotalPages }}</span></a></li>
-          {{ end }}
+      {{ range .Pages }}
+        {{ if eq . $.CurrentPage }}
+          <li class="moj-pagination__item moj-pagination__item--active"><span class="govuk-visually-hidden">Page </span>{{ . }}<span class="govuk-visually-hidden"> of {{ $.TotalPages }}</span></li>
+        {{ else }}
+          <li class="moj-pagination__item"><a class="moj-pagination__link" href="{{ $.Query }}page={{ . }}"><span class="govuk-visually-hidden">Page </span>{{ . }}<span class="govuk-visually-hidden"> of {{ $.TotalPages }}</span></a></li>
         {{ end }}
+      {{ end }}
 
-        {{ if .HasNext }}
-          <li class="moj-pagination__item  moj-pagination__item--next">
-            <a class="moj-pagination__link" href="?page={{ .NextPage }}">Next<span class="govuk-visually-hidden"> page</span></a>
-          </li>
-        {{ end }}
-      </ul>
-    {{ end }}
+      {{ if .HasNext }}
+        <li class="moj-pagination__item  moj-pagination__item--next">
+          <a class="moj-pagination__link" href="{{ .Query }}page={{ .NextPage }}">Next<span class="govuk-visually-hidden"> page</span></a>
+        </li>
+      {{ end }}
+    </ul>
+  {{ end }}
+{{ end }}
+
+{{ define "pagination" }}
+  <nav class="moj-pagination" id="pagination-label">
+    <p class="govuk-visually-hidden" aria-labelledby="pagination-label">Pagination navigation</p>
```

Comment:
I know this is straight out of the docs, but I think this is cleaner (it certainly works better with a screenreader):
```suggestion
  <nav class="moj-pagination" aria-label="Pagination navigation">
```
Otherwise we're saying that the `p` is labelled by the entire contents of its parent, which is nonsense.

This gets rid the duplicate ID situation, but does adds another problem in that `aria-label` needs to be unique. So the one in the footer should probably be the slightly clunky "Pagination navigation (in footer)"

Reply:
🤦  Oh I messed that one up, but fixed the duplicate one